src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.IllegalDeadlineException;
+import duke.exceptions.IllegalEventException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import static duke.constants.CommandConstants.COMMAND_ADD_TODO;
+import static duke.constants.CommandConstants.COMMAND_ADD_EVENT;
+import static duke.constants.CommandConstants.COMMAND_ADD_DEADLINE;
+import static duke.constants.CommandConstants.COMMAND_MARK_DONE;
+import static duke.constants.CommandConstants.COMMAND_LIST;
+import static duke.constants.CommandConstants.COMMAND_EXIT;
+import static duke.constants.CommandConstants.COMMAND_DELETE;
+
+import static duke.constants.TaskConstants.PREFIX_BY_DEADLINE;
+import static duke.constants.TaskConstants.PREFIX_TIME_EVENT;
+import static duke.constants.Messages.HORIZONTAL_LINE;
+
+import static duke.constants.Messages.MESSAGE_NO_TASK;
+
+import java.util.Scanner;
+
+public class Duke {
+
+    private static Ui ui = new Ui();
+    private static ExceptionHandler exceptionHandler = new ExceptionHandler();
+    private static Storage storage = new Storage();
+    private static Parser parser = new Parser();
+
+    private static TaskList tasks = new TaskList();
+
+    public static void main(String[] args) throws IOException {
+        ui.printGreeting();
+        readAndExecuteCommand();
+    }
+
+    /**
+     * Reads and executes command from user inputs.
+     */
+    private static void readAndExecuteCommand() throws IOException {
+        Scanner in = new Scanner(System.in);
+        tasks.tasks = storage.readDataFromFile();
+        while (true) {
+            String userInput = getUserInput(in);
+            try {
+                executeCommand(userInput);
+            } catch (DukeException e) {
+                exceptionHandler.handleDukeExceptions(e);
+            } catch (Exception e) {
+                exceptionHandler.handleOtherExceptions(e);
+            }
+        }
+    }
+
+    /**
+     * Reads the text entered by the user.
+     *
+     * @param in The Scanner to read user input.
+     * @return line Full line entered by the user.
+     */
+    private static String getUserInput(Scanner in) {
+        String line = in.nextLine();
+        return line;
+    }
+
+    /**
+     * Executes the command as specified by the user input.
+     *
+     * @param userInput Raw input from user.
+     */
+    private static void executeCommand(String userInput) throws DukeException, IOException {
+        final String[] commandTypeAndParams = parser.splitCommandAndArgs(userInput);
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+
+        switch (commandType) {
```

Comment:
Good use of switch case, and abstraction of each command into separate functions!